src/status_im/models/wallet.cljs

```diff
@@ -216,3 +242,17 @@
                                (clear-error-message :balance-update)
                                (assoc-in [:wallet :balance-loading?] true)
                                (assoc :prices-loading? true))})))
+
+(defn open-modal-wallet-for-trasnaction [db transaction tx-object]
+  (let [{:keys [gas gas-price]} transaction
+        {:keys [wallet-set-up-passed?]} (:account/account db)]
+    {:db         (assoc-in db [:wallet :send-transaction] transaction)
+     :dispatch-n [[:update-wallet]
+                  (when-not gas
+                    [:wallet/update-estimated-gas tx-object])
+                  (when-not gas-price
+                    [:wallet/update-gas-price])
+                  [:navigate-to
+                   (if wallet-set-up-passed?
+                     :wallet-send-modal-stack
+                     :wallet-send-modal-stack-with-onboarding)]]}))
```

Comment:
Please encode logic with `fx/merge` macro and model functions instead of `:dispatch-n`, there is a `status-im.models.wallet/update-wallet` fn already for example.
In worst case, when we don't want to do any refactoring in the PR and some functionality is only exposed via events, we can at least do:
```clj
(fx/merge cofx
          {:dispatch-n collection-of-events-which-have-no-model-functions-yet}
          model-fn-1
          model-fn-2
          ...)
```

By doing that, it will be clear that we are trying to rid of the `:dispatch-n` pattern + it will be super easy to actually erase it completely once those events are refactored into model functions.

Reply:
yeah, i expected such comment , you're right, but i don't want to made such change in this PR, let me create separate PR for this

Reply:
Sure, makes sense !